econml/_ortho_learner.py

```diff
@@ -540,15 +558,70 @@ def fit(self, Y, T, X=None, W=None, Z=None, *, sample_weight=None, sample_var=No
         Y, T, X, W, Z, sample_weight, sample_var, groups = check_input_arrays(
             Y, T, X, W, Z, sample_weight, sample_var, groups)
         self._check_input_dims(Y, T, X, W, Z, sample_weight, sample_var, groups)
-        nuisances, fitted_inds = self._fit_nuisances(Y, T, X, W, Z, sample_weight=sample_weight, groups=groups)
-        self._fit_final(self._subinds_check_none(Y, fitted_inds),
-                        self._subinds_check_none(T, fitted_inds),
-                        X=self._subinds_check_none(X, fitted_inds),
-                        W=self._subinds_check_none(W, fitted_inds),
-                        Z=self._subinds_check_none(Z, fitted_inds),
-                        nuisances=tuple([self._subinds_check_none(nuis, fitted_inds) for nuis in nuisances]),
-                        sample_weight=self._subinds_check_none(sample_weight, fitted_inds),
-                        sample_var=self._subinds_check_none(sample_var, fitted_inds))
+
+        all_nuisances = []
+        fitted_inds = None
+
+        for _ in range(monte_carlo_iterations or 1):
+            nuisances, new_inds = self._fit_nuisances(Y, T, X, W, Z, sample_weight=sample_weight, groups=groups)
```

Comment:
If we fit multiple times, what version is models_nuisance storing? Is it storing for each mc iteration and for each fold in the kfold, the fitted model? Or is it storing only the final version of mc iterations? The latter would be inconsistent.

Similarly, when we call "score" we use the fitted models_nuisance to predict. But which fitted models_nusiance? just the last mc iter? or all the mc_iters?

Reply:
Right now, it is fitting only a single model multiple times and retaining information from the latest run (because users might expect a single model per fold, not one per fold*mc_iter combination), but I can change that.

Perhaps best would be to have a model wrapper that wraps all of the individual ones but provides a single prediction, similar to bootstrap?

Reply:
Actually I take it back: `_crossfit` is cloning the model, so we don't call fit repeatedly on the same instance.